frontend/src/modules/2DViewer/layers/components/utils.tsx

```diff
@@ -0,0 +1,93 @@
+import { SortableListItemProps } from "@lib/components/SortableList";
+
+import { ColorScaleComponent } from "./ColorScaleComponent";
+import { DeltaSurfaceComponent } from "./DeltaSurfaceComponent";
+import { LayerComponent } from "./LayerComponent";
+import { LayersActionGroup } from "./LayersActions";
+import { SettingsGroupComponent } from "./SettingsGroupComponent";
+import { SharedSettingComponent } from "./SharedSettingComponent";
+import { ViewComponent } from "./ViewComponent";
+
+import { ColorScale } from "../ColorScale";
+import { DeltaSurface } from "../DeltaSurface";
+import { SettingsGroup } from "../SettingsGroup";
+import { SharedSetting } from "../SharedSetting";
+import { View } from "../View";
+import { Group, Item, instanceofGroup, instanceofLayer } from "../interfaces";
+
+export function makeComponent(
+    item: Item,
+    layerActions?: LayersActionGroup[],
+    onActionClick?: (identifier: string, group: Group) => void
+): React.ReactElement<SortableListItemProps> {
+    if (instanceofLayer(item)) {
+        return <LayerComponent key={item.getItemDelegate().getId()} layer={item} />;
+    }
+    if (instanceofGroup(item)) {
+        if (item instanceof SettingsGroup) {
+            return (
+                <SettingsGroupComponent
+                    key={item.getItemDelegate().getId()}
+                    group={item}
+                    actions={layerActions}
+                    onActionClick={onActionClick}
+                />
+            );
+        } else if (item instanceof View) {
+            return (
+                <ViewComponent
+                    key={item.getItemDelegate().getId()}
+                    group={item}
+                    actions={layerActions ? filterAwayViewActions(layerActions) : undefined}
+                    onActionClick={onActionClick}
+                />
+            );
+        } else if (item instanceof DeltaSurface) {
+            return (
+                <DeltaSurfaceComponent
+                    key={item.getItemDelegate().getId()}
+                    deltaSurface={item}
+                    actions={layerActions ? filterAwayNonSurfaceActions(layerActions) : undefined}
+                    onActionClick={onActionClick}
+                />
+            );
+        }
+    }
```

Comment:
The `instanceofGroup` check is actually not necessary - so I'm gonna remove it. I am actually a fan of early returns instead of `else if` as well, so don't know what happened here - gonna remove them.